adlchecker/src/main/java/com/nedap/archie/adlchecker/TerminologyContentGenerator.java

```diff
@@ -140,6 +170,28 @@ public boolean terminologyHasCode(Archetype archetype, CObject cObject) {
         return true;
     }
 
+    /**
+     * Check if terminology has a code. Use only for at- and ac-codes, NOT For id codes!!
+     * @param archetype
+     * @param code
+     * @return
+     */
+    public boolean terminologyHasCodeForAllLanguages(Archetype archetype, String code) {
+        if(AOMUtils.isIdCode(code)) {
+            throw new IllegalArgumentException("this method only checks at- and ac- codes, not id codes");
+        }
+        int codeSpecializationDepth = AOMUtils.getSpecializationDepthFromCode(code);
```

Comment:
This method mostly contains duplicate code from terminologyHasCodeForAllLanguages. Maybe extract this code or something?

Reply:
except that it's not and it does something different and extracting the code would make it much harder to read